### ISSUE TITLE
WARNING - ERR DRIVER-NOT-CONNECTED

### ISSUE THREAD

**Issue by EbberBebber** [kantlivelong/OctoPrint-UPS#15]
This UPS code works without any problems for me, but when USB cable is not connected it records warning message to the Octoprint.log continuously every second filling up the log memory. It does not effect to printers operation. 
  Is this meant to be this way or is there in the code some setting to disable this warning?

2024-02-06 11:14:33,355 - octoprint.plugins.ups - WARNING - ERR DRIVER-NOT-CONNECTED
2024-02-06 11:14:34,396 - octoprint.plugins.ups - WARNING - ERR DRIVER-NOT-CONNECTED
2024-02-06 11:14:35,437 - octoprint.plugins.ups - WARNING - ERR DRIVER-NOT-CONNECTED
2024-02-06 11:14:36,589 - octoprint.plugins.ups - WARNING - ERR DRIVER-NOT-CONNECTED
2024-02-06 11:14:37,677 - octoprint.plugins.ups - WARNING - ERR DRIVER-NOT-CONNECTED
2024-02-06 11:14:38,723 - octoprint.plugins.ups - WARNING - ERR DRIVER-NOT-CONNECTED
2024-02-06 11:14:39,764 - octoprint.plugins.ups - WARNING - ERR DRIVER-NOT-CONNECTED
2024-02-06 11:14:40,805 - octoprint.plugins.ups - WARNING - ERR DRIVER-NOT-CONNECTED
2024-02-06 11:14:41,828 - octoprint.plugins.ups - WARNING - ERR DRIVER-NOT-CONNECTED

**Comment by kantlivelong** [comment]
Could be worth adding some noise filtering. Other than that I would suggest disabling the plugin when the UPS isn't connected.

**Comment by EbberBebber** [comment]
Some how I started to get following error messages every second while the UPS is connected to the Octoprint. Maybe I messed up with a setting or something.

2024-02-15 07:12:24,121 - octoprint.plugins.ups - ERROR - An exception occurred while getting vars info
Traceback (most recent call last):
  File "/home/pi/oprint/lib/python3.9/site-packages/octoprint_ups/__init__.py", line 132, in _loop
    vars = self.ups.list_vars(ups=self.config['ups'])
  File "/home/pi/oprint/lib/python3.9/site-packages/nut2.py", line 157, in list_vars
    self._srv_handler.write(b"LIST VAR %s\n" % ups.encode('utf-8'))
AttributeError: 'NoneType' object has no attribute 'encode'
2024-02-15 07:12:25,182 - octoprint.plugins.ups - ERROR - An exception occurred while getting vars info
Traceback (most recent call last):
  File "/home/pi/oprint/lib/python3.9/site-packages/octoprint_ups/__init__.py", line 132, in _loop
    vars = self.ups.list_vars(ups=self.config['ups'])
  File "/home/pi/oprint/lib/python3.9/site-packages/nut2.py", line 157, in list_vars
    self._srv_handler.write(b"LIST VAR %s\n" % ups.encode('utf-8'))
AttributeError: 'NoneType' object has no attribute 'encode'
2024-02-15 07:12:26,251 - octoprint.plugins.ups - ERROR - An exception occurred while getting vars info
Traceback (most recent call last):
  File "/home/pi/oprint/lib/python3.9/site-packages/octoprint_ups/__init__.py", line 132, in _loop
    vars = self.ups.list_vars(ups=self.config['ups'])
  File "/home/pi/oprint/lib/python3.9/site-packages/nut2.py", line 157, in list_vars
    self._srv_handler.write(b"LIST VAR %s\n" % ups.encode('utf-8'))
AttributeError: 'NoneType' object has no attribute 'encode'
2024-02-15 07:12:27,315 - octoprint.plugins.ups - ERROR - An exception occurred while getting vars info
Traceback (most recent call last):

After this I set up logging for the UPS and again it is repeating same data out every second. Is there some setting to slow down the rate of messaging?

2024-02-15 10:07:45,476 - octoprint.plugins.ups - DEBUG - {'battery.charge': '100', 'battery.charge.low': '20', 'battery.runtime': '8437', 'battery.type': 'PbAc', 'device.mfr': 'EATON', 'device.model': 'Ellipse PRO 1500 ', 'device.serial': 'P344M17EJ7', 'device.type': 'ups', 'driver.name': 'usbhid-ups', 'driver.parameter.pollfreq': '30', 'driver.parameter.pollinterval': '1', 'driver.parameter.port': 'auto', 'driver.parameter.synchronous': 'no', 'driver.version': '2.7.4', 'driver.version.data': 'MGE HID 1.40', 'driver.version.internal': '0.41', 'input.frequency': '60.0', 'input.transfer.high': '138', 'input.transfer.low': '93', 'input.voltage': '125.0', 'input.voltage.extended': 'no', 'outlet.1.desc': 'PowerShare Outlet 1', 'outlet.1.id': '2', 'outlet.1.status': 'on', 'outlet.1.switchable': 'no', 'outlet.2.desc': 'PowerShare Outlet 2', 'outlet.2.id': '3', 'outlet.2.status': 'on', 'outlet.2.switchable': 'no', 'outlet.desc': 'Main Outlet', 'outlet.id': '1', 'outlet.switchable': 'no', 'output.frequency': '60.0', 'output.frequency.nominal': '60', 'output.voltage': '126.0', 'output.voltage.nominal': '115', 'ups.beeper.status': 'enabled', 'ups.delay.shutdown': '20', 'ups.delay.start': '30', 'ups.firmware': '01.16.0020', 'ups.load': '5', 'ups.mfr': 'EATON', 'ups.model': 'Ellipse PRO 1500 ', 'ups.power': '82', 'ups.power.nominal': '1500', 'ups.productid': 'ffff', 'ups.realpower': '0', 'ups.serial': 'P344M17EJ7', 'ups.status': 'OL', 'ups.timer.shutdown': '-1', 'ups.timer.start': '-1', 'ups.vendorid': '0463'}
2024-02-15 10:07:46,536 - octoprint.plugins.ups - DEBUG - {'battery.charge': '100', 'battery.charge.low': '20', 'battery.runtime': '8437', 'battery.type': 'PbAc', 'device.mfr': 'EATON', 'device.model': 'Ellipse PRO 1500 ', 'device.serial': 'P344M17EJ7', 'device.type': 'ups', 'driver.name': 'usbhid-ups', 'driver.parameter.pollfreq': '30', 'driver.parameter.pollinterval': '1', 'driver.parameter.port': 'auto', 'driver.parameter.synchronous': 'no', 'driver.version': '2.7.4', 'driver.version.data': 'MGE HID 1.40', 'driver.version.internal': '0.41', 'input.frequency': '60.0', 'input.transfer.high': '138', 'input.transfer.low': '93', 'input.voltage': '125.0', 'input.voltage.extended': 'no', 'outlet.1.desc': 'PowerShare Outlet 1', 'outlet.1.id': '2', 'outlet.1.status': 'on', 'outlet.1.switchable': 'no', 'outlet.2.desc': 'PowerShare Outlet 2', 'outlet.2.id': '3', 'outlet.2.status': 'on', 'outlet.2.switchable': 'no', 'outlet.desc': 'Main Outlet', 'outlet.id': '1', 'outlet.switchable': 'no', 'output.frequency': '60.0', 'output.frequency.nominal': '60', 'output.voltage': '126.0', 'output.voltage.nominal': '115', 'ups.beeper.status': 'enabled', 'ups.delay.shutdown': '20', 'ups.delay.start': '30', 'ups.firmware': '01.16.0020', 'ups.load': '5', 'ups.mfr': 'EATON', 'ups.model': 'Ellipse PRO 1500 ', 'ups.power': '82', 'ups.power.nominal': '1500', 'ups.productid': 'ffff', 'ups.realpower': '0', 'ups.serial': 'P344M17EJ7', 'ups.status': 'OL', 'ups.timer.shutdown': '-1', 'ups.timer.start': '-1', 'ups.vendorid': '0463'}
2024-02-15 10:07:47,596 - octoprint.plugins.ups - DEBUG - {'battery.charge': '100', 'battery.charge.low': '20', 'battery.runtime': '8437', 'battery.type': 'PbAc', 'device.mfr': 'EATON', 'device.model': 'Ellipse PRO 1500 ', 'device.serial': 'P344M17EJ7', 'device.type': 'ups', 'driver.name': 'usbhid-ups', 'driver.parameter.pollfreq': '30', 'driver.parameter.pollinterval': '1', 'driver.parameter.port': 'auto', 'driver.parameter.synchronous': 'no', 'driver.version': '2.7.4', 'driver.version.data': 'MGE HID 1.40', 'driver.version.internal': '0.41', 'input.frequency': '60.0', 'input.transfer.high': '138', 'input.transfer.low': '93', 'input.voltage': '125.0', 'input.voltage.extended': 'no', 'outlet.1.desc': 'PowerShare Outlet 1', 'outlet.1.id': '2', 'outlet.1.status': 'on', 'outlet.1.switchable': 'no', 'outlet.2.desc': 'PowerShare Outlet 2', 'outlet.2.id': '3', 'outlet.2.status': 'on', 'outlet.2.switchable': 'no', 'outlet.desc': 'Main Outlet', 'outlet.id': '1', 'outlet.switchable': 'no', 'output.frequency': '60.0', 'output.frequency.nominal': '60', 'output.voltage': '126.0', 'output.voltage.nominal': '115', 'ups.beeper.status': 'enabled', 'ups.delay.shutdown': '20', 'ups.delay.start': '30', 'ups.firmware': '01.16.0020', 'ups.load': '5', 'ups.mfr': 'EATON', 'ups.model': 'Ellipse PRO 1500 ', 'ups.power': '82', 'ups.power.nominal': '1500', 'ups.productid': 'ffff', 'ups.realpower': '0', 'ups.serial': 'P344M17EJ7', 'ups.status': 'OL', 'ups.timer.shutdown': '-1', 'ups.timer.start': '-1', 'ups.vendorid': '0463'}
